File: back/src/main/java/com/cherry/erp/company/model/persistent/Company.java
package com.cherry.erp.company.model.persistent;

import com.cherry.erp.common.model.persistent.GenericEntity;
import com.cherry.erp.company.model.enums.CompanyOrganisationTypeEnum;
import com.cherry.erp.company.model.enums.CompanySizeEnum;
import com.cherry.erp.company.model.enums.SupportEnum;
import com.vladmihalcea.hibernate.type.json.JsonBinaryType;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.SQLDelete;
import org.hibernate.annotations.Type;
import org.hibernate.annotations.TypeDef;
import org.hibernate.annotations.Where;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "COMPANIES")
@SequenceGenerator(name = "SEQ", sequenceName = "companies_seq", allocationSize = 1)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@TypeDef(name = "jsonb", typeClass = JsonBinaryType.class)
@Where(clause = "is_active = true")
@SQLDelete(sql = "UPDATE COMPANIES SET is_active = false WHERE id = ? AND version = ?")
public class Company extends GenericEntity {

    private static final long serialVersionUID = 1090088612933845086L;

    @Column(name = "NAME", nullable = false)
    private String name;

    @Column(name = "SECTOR")
    private String sector;

    @Column(name = "PAYS")
    private String pays;

    @Enumerated(EnumType.STRING)
    private CompanySizeEnum size;

    @Enumerated(EnumType.STRING)
    private CompanyOrganisationTypeEnum type;

    @Column(name = "LOGO")
    @Type(type = "text")
    private String logo;

    @Column(name = "slogan")
    private String slogan;

    @Column(name = "ADRESSE")
    private String adresse;

    @Column(name = "REGISTRE_DE_COMMERCE")
    private String registreDeCommerce;

    @Column(name = "MATRICULE_FISCALE")
    private String matriculeFiscale;

    @Column(name = "CODE_DOUANE")
    private String codeDouane;

    @Column(name = "EMAIL")
    private String email;

    @Column(name = "TEL")
    private String tel;

    @Column(name = "FAX")
    private String fax;

    @Column(name = "MOBILE")
    private String mobile;

    @Column(name = "SUPPORT")
    private SupportEnum support; //type de support: silver, gold, platinum

    @Column(name = "language")
    private String language;

    @Column(name = "theme")
    private String theme;

    @Column(name = "localize")
    private Boolean localize;

    @Column(name = "payroll_in_progress")
    private Boolean payrollInProgress;

}

File: front/src/app/shared/models/company.ts

import {GenericEntity} from './genericEntity';

export class CompanySetting {
  name: string;
  value: string;
  description: string;

  constructor(name: string, value: any) {
      this.name = name;
      this.value = value;
  }
}

export class CompanySettings {
  settings: CompanySetting[];
  signatureImage: string;
}

export class DocumentTax {
  name: string;
  defaultValue: number;
  positiveValue = true;
}

export class Company extends GenericEntity {

  name: string;
  logo: string;
  slogan: string;
  adresse: string;
  registreDeCommerce: string;
  matriculeFiscale: string;
  codeDouane: string;
  email: string;
  tel: string;
  fax: string;
  mobile: string;
  taxes: any[];
  documentTaxes: DocumentTax[];
  currencies: any[];
  license: any;
  settings: CompanySettings;
  support: any;
  language: string;
  theme: string;
  localize: boolean;

  businessDays: any;

  constructor(values: object = {}) {
    super();
    Object.assign(this, values);
  }
}
